### PR TITLE
Add section to show the active PHP version

### LINKED ISSUE #67: Implement a prompt section for PHP version (🐘)
## Feature Request

**Is your feature request related to a problem? Please describe.**
This feature is intended to achieve feature parity with the following Spaceship section file: https://github.com/denysdovhan/spaceship-prompt/blob/master/sections/php.zsh

**Describe the solution you'd like**
Porting over the existing zsh implementation of the PHP section file, showing the current PHP version when the prompt is in the same directory as a `*.php` or `composer.json` file.

### PR DESCRIPTION
**Pull request by matchai**
## Description

Implement Rust version section based on the implementation on spaceship.
- Show version section when `composer.json` or `*.php` are present
- Write tests for the new section
- Update documentation with the new section

## Motivation and Context

Closes #67 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots (if appropriate):
image

## How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**

## Checklist:

- [x] I have checked that no other PR duplicates mine
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: tests/__sf_section_rust.test.fish [hunk 1]
```diff
@@ -47,7 +47,7 @@ end
 test "Doesn't print the section when Cargo.toml and *.rs aren't present"
 	(
 		rm -rf /tmp/tmp-spacefish/Cargo.toml
-	) = (__sf_section_golang)
+	) = (__sf_section_rust)
```
Comment (matchai) [inline tests/__sf_section_rust.test.fish:50]: Noticed a little slip-up in the rust tests.
A little drive-by fix. 🚗

Comment (Snuggle) [reply tests/__sf_section_rust.test.fish:50]: Interesting how this didn't come up with an error or something.
Nice spot! :eyes:

File: functions/__sf_section_php.fish [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	if not test -f composer.json \
+		-o (count *.php) -gt 0
+		return
+	end
+
+	if not type -q php
+		return
+	end
+
+	set -l php_version (php -v | string match -r 'PHP\s*[0-9.]+' | string split ' ')[2]
```
Comment (Snuggle) [inline functions/__sf_section_php.fish:33]: I like how you've continued to use Fish's string function here!

File: tests/__sf_section_php.test.fish [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		set_color --bold fff
+		echo -n " "
+		set_color normal
+	) = (__sf_section_php)
+end
+
+test "Changing SPACEFISH_PHP_PREFIX changes the character prefix"
+	(
+		set sf_exit_code 0
+		set SPACEFISH_PHP_PREFIX ·
```
Comment (Snuggle) [inline tests/__sf_section_php.test.fish:74]: You haven't used quotation marks here or on line 90, but you used them on line 56. Which should be preferred?

File: functions/__sf_section_php.fish [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	# ------------------------------------------------------------------------------
+
+	[ $SPACEFISH_PHP_SHOW = false ]; and return
+
+	if not test -f composer.json \
+		-o (count *.php) -gt 0
+		return
+	end
+
+	if not type -q php
```
Comment (Snuggle) [inline functions/__sf_section_php.fish]: Wouldn't this be better to have before the test above? I don't have PHP installed, but it will always check for `composer.json` or a `*.php` file regardless. This seems inefficient performance-wise to me.

**Review by Snuggle — CHANGES_REQUESTED** [review]

File: functions/__sf_section_php.fish (hunk 4 above)
Comment (Snuggle) [reply functions/__sf_section_php.fish]: Logically, it makes a bit more sense to check if PHP is installed _before_ checking the entire current directory for PHP-related files. Might be worth checking other sections to see if they [follow this guideline](https://github.com/matchai/spacefish/blob/master/CONTRIBUTING.md#will-it-slow-down-the-prompt), in the same way.

File: tests/__sf_section_php.test.fish (hunk 3 above)
Comment (matchai) [reply tests/__sf_section_php.test.fish:74]: On line 90, there are quotes because the string is followed by a space.

File: functions/__sf_section_php.fish (hunk 4 above)
Comment (matchai) [reply functions/__sf_section_php.fish]: Makes sense. I'll go ahead and fix that 👍

**Comment by matchai** [comment]
I have a PR lined up after this one fixing some consistency in sections that you mentioned above. 👍

**Review by Snuggle — APPROVED** [review]
